### PR TITLE
WJohn's Hyper Customizable Lizard Sprites

### PR DESCRIPTION
**Pull request by Incoming5643**
All of WJohn's Lizard Sprites are finally in game! (Besides the wings)

Each portion individually selectable upon character creation, the number of lizard combinations is bordering on absurdist with six customizable fields.

36,864 unique combinations!
(309,237,645,312 If you count color variations)

Never feel like you can't be unique again! Have fun looking fabulous as the crew beats your special little skull in!

d10_icon

Thing to do:
- [x] fix the conflict(s) 
- [x] finalize the coverage flag behavior
- [x] code beautification
- [x] make lizard features play ball with genetics so lings don't get screwed 
- [x] free lizard_parts from the flaky hell of being both human defined and dna defined

### CODE REVIEW

**Comment by Incoming5643** [comment]
Conflicts with tail wagging. It'd be a herculean effort to make them work together in terms of sprites

Full disclaimer some of the code is pretty raw looking right now, but goddamn if it doesn't all WORK.

Some coverage flags might act a little weird until I catch them all.

File: code/modules/client/preferences_savefile.dm [hunk 1]
```diff
@@ -187,6 +187,12 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	S["socks"]				>> socks
 	S["backbag"]			>> backbag
 	S["mutant_color"]		>> mutant_color
+	S["lizard_tail"]		>> lizard_tail
```
Comment (RemieRichards) [inline code/modules/client/preferences_savefile.dm]: pretty sure you can store all this junk as a list.

**Comment by PKPenguin321** [comment]
funny how all the snowflake stuff appears on the github *after* the freeze ends

**Comment by WJohn** [comment]
IT'S HAPPENING

AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by Iamgoofball** [comment]
AAAAAAAAAAAAAAAAAA

BURNIT BURNIT BURNIT BURNIT

GAS THE LIZARDS

**Comment by palpatine213** [comment]
Quick. We must nerf the lizards to balance out the snowflake!

File: code/modules/client/preferences_savefile.dm (hunk 1 above)
Comment (Incoming5643) [inline code/modules/client/preferences_savefile.dm]: Yeah code beautification is definitely still pending on this one

**Comment by Stickymayhem** [comment]
this is missing the FUCK tag

**Comment by Fox-McCloud** [comment]
Big :+1:  from me!

**Comment by GunHog** [comment]
This is what I have to say to this: https://youtu.be/W6x9l4W_g6A?t=56

**Comment by freerealestate** [comment]
Never played a lizard. I like it. Hope it pisses people off.

**Comment by Dorsisdwarf** [comment]
Damnit. Now my "unzip revolver on any lizard I catch who begged an admin for wj's awful stupid sprites" strategy doesn't work at all!

**Comment by Miauw62** [comment]
>thousands of unique sprites and combinations
>a good thing

**Comment by Miauw62** [comment]
and by that i mean as if disguising as somebody wasnt hard enough already!

**Comment by Incoming5643** [comment]
All these bits and bobs have hiding flags related to clothing that can make pretending to be them easier, a guy in a biosuit is still just a guy in a biosuit.

That said there's already a more practical problem as disguising as a lizard as a nonlizard in that lizards use the "hiss" verb when they talk, which humans have no way to replicate. This can humorously be overcome by only talking in questions and exclamations!!

**Comment by SconesC** [comment]
Wow, that's actually really cool. Glad to see these have finally moved on to a feature and not WJohn applying them every round.

**Comment by Kromgar** [comment]
Fuck this shit. 

I hate the shit like wings and the thing hanging off heads. I would be fine with customizable ears and hair frills or whatever they are called

**Comment by Incoming5643** [comment]
Wings aren't in this specifically because they were a bit over the top.

The zones are tail, horns (hair), frills (ears), snouts (nose), spines (back), and body markings.

The angler lure (classed as a horn type) is pretty much the silliest looking thing in the pull.

**Comment by SconesC** [comment]
The wings are autistic and I'm glad they're gone. The hair that I've been seeing lately is similarly fuck tier.

**Comment by Incoming5643** [comment]
if anyone wants a closer look about what physical features in the pull look like here's the sprites @WJohn sent to me before I had to slice them all up for proper layering http://download1478.mediafire.com/7iuv9id96ong/ptv1fcnj5ajt540/Character+Creation.rar

**Comment by Nienhaus** [comment]
@Miauw62 Why'd you remove the sprites tag?

**Comment by PKPenguin321** [comment]
I think the sprite tag is supposed to only be for issue reports

**Comment by GunHog** [comment]
It is for sprite PRs as well, to bring attention to the art maintainers (WJohn, Nienhaus). The sprites tag is in fact appropriate for this PR, as this is a sprite based PR.

**Comment by PKPenguin321** [comment]
\>go to https://github.com/tgstation/-tg-station/labels/Sprites
\>click issues, see issues labeled "Sprites"
\>click Pull Requests, see that there are no PRs labeled "Sprites"

it's just for issues

**Comment by Xhuis** [comment]
https://github.com/tgstation/-tg-station/pull/8198
nah

**Comment by Nienhaus** [comment]
I thought it was going to be used so WJohn and I know what PRs have sprites to look at.

**Comment by duncathan** [comment]
@PKPenguin321 Yeah, there are no open PRs with the sprites tag. Take a look at the 93 closed ones, maybe.

**Comment by PKPenguin321** [comment]
rip me

**Comment by Nienhaus** [comment]
And that is why I asked why it was removed.

**Comment by RemieRichards** [comment]
Really love how you "forgot" to say you reverted tail wagging anywhere except the commit, kudos.

**Comment by Fox-McCloud** [comment]
@RemieRichards 

Wait? What? Really.

@Incoming5643

:-1:  x :100:

Tail wagging is the best thing in the world. Don'tchu dare touch it.

**Comment by RemieRichards** [comment]
@Fox-McCloud Well prior to this WJ and Incoming were both complaining about tail wagging preventing there ever being tail clothing (Something which realistically was never going to happen) so instead there's now this, something which achieves the same goal of removing it.

Realistically the only part of this that clashes is spikes on tails and custom tail shapes,
The shapes can be solved by adding an animation for each shape (If WJ's produced enough sprites for 36,864 you really can't complain about somewhere around 5 tail animations)
The spikes can be solved with one animation per spike shape, see above.

So "As a matter of necessity" is complete codswallop.

**Comment by Dorsisdwarf** [comment]
Tail wagging is not only a bit stupid, it's the reason I actually dislike you, genuinely not the "lol worst coder", for doing it instead of finishing your stupid dismemberment. Along with all your other stupid projects, here and elsewhere, especially as it was "almost ready" six frakking months ago. Kill it or finish it, but stop doing moronic lizardhugging changes while everyone's waiting for your actual project

**Comment by RemieRichards** [comment]
@Dorsisdwarf "Fuck off"

**Comment by Iamgoofball** [comment]
@RemieRichards "Dismemberment when"

**Comment by PKPenguin321** [comment]
wow holy shit the amount of salt over some dumbass lizard textures
chill guys

**Comment by RogerWilcoBeep** [comment]
Christ.
/tg/station's really fell, huh.

**Comment by Incoming5643** [comment]
I wasn't trying to stealth it, I warned more than once that the wagging tail thing wouldn't work with custom sprite without a ton of sprite work.

Hell I even left it out of the initial commit for this because I was hoping I could find some way to make it work, but I did the math here: 

260 new frames of animation to make this work with tail wagging.
This whole pull has 130 frames right now.

Two thirds of this effort would be just to make sure lizards can wag their tails. Honestly it doesn't feel worth it, especially when John's been making these sprites just for our players and the tail wagging thing is a modified port.

Also I'd just like to point out that the players most likely to use tail wagging in game, hardcore lizard roleplayers, are also the same people who John's been manually giving these sprites to. Giving them those sprites requires them to be reskinned humans, which means that they couldn't use tail wagging either.

So in short, I'm not trying to slight anyone here, it's just how the dice fell. I'm sorry. Seriously.

It's just lizards.

**Comment by Incoming5643** [comment]
(also "tail clothes" are dumb and I never thought they were something that would [or should] be added, the whole point of covering outfits is to HIDE the lizardness guys)

**Comment by Razharas** [comment]
I dont like lizards at all, imagine for much better the world of ss13 would be without them

**Comment by demonfiren** [comment]
I can hiss to this.

**Comment by Loonicus** [comment]
So when will humans become more customizable?

**Comment by Incoming5643** [comment]
As soon as sprites for it really, the system is generalized enough.

**Comment by GunHog** [comment]
MERGE PLZ, EVEN THOUGH WE WILL LOSE WAGGING, THIS IS BETTER

**Comment by demonfiren** [comment]
Yes, do it. I don't care about waggotry.

**Comment by RemieRichards** [comment]
I port you guys something nice and you spit in my face.
Wonderful.

**Comment by Fox-McCloud** [comment]
removing tail wagging in this PR instead of integrating it into this system.

Very shitty.

:-1:

**Comment by GunHog** [comment]
@RemieRichards I love tail wagging, I want to keep it, but not at the cost of these wonderful PR! I have wanted to be horny again ever since Paprika removed them! If there were some way we could keep both features...

**Comment by RemieRichards** [comment]
There is, I've explained it, it's laziness/bias on the part of the spriter and dev.

**Comment by Incoming5643** [comment]
It's not like we couldn't re-add it later or anything if someone put the effort in. I mean if it's removed and people are suddenly up in arms about wagging tails I'm sure someone would get the drive to sprite it.

Also don't merge this till I fix the doube defined human/dna nature of the system right now, it's functional but hacky. Needs to be made full DNA, and probably merged with mcolor.

**Comment by Incoming5643** [comment]
With the latest commit I deem this feature complete, the lizard sprites successfully persist through the usual gamut of identity shenanigans (cloning, ling business, pod people, etc) and are implemented in a way that's about as reasonable as can be without rewriting dna.

As I've said if someone can provide the sprites for wagging tails for the new tails and the new spines I will gladly append the commit to retain the feature. I just don't have the ability to do it myself.

If this pull is merged before #9880 there will be a small conflict to resolve for that pull, or visa versa if his gets pulled first (it's a good fix).

**Comment by WJohn** [comment]
I didn't think you'd be so upset on losing tail wagging, but I don't want to feel like an asshole over it. I'll go sprite them to work with these.

**Comment by GunHog** [comment]
<b>WHEEEEEEEEEEEEEEEEEEEEEEEEEE!!!</b> We get both! Praise the Xeno master!

**Comment by demonfiren** [comment]
If you're doing the spriting, then please ensure that the sideview also produces side-wagging, because I still believe the up-down sprites are mildly more awful than the left-right ones.

**Comment by WJohn** [comment]
Already done that. I've got 6/13 of them done right now but it's a really time consuming process. It's 9 frames on each side, and while left/right is interchangable and front/back is mostly the same it's still a lot to do. Should be done soon-ish, probably by tomorrow.

**Comment by Miauw62** [comment]
[snowflake intensifies]

**Comment by GunHog** [comment]
This is why I find WJohn to be one of the best contributors for /tg/. Very few of us would go through the enormous effort to work around something like this. This guy deserves a medal and a free game on Steam.

**Comment by WJohn** [comment]
Here you go:
http://www.mediafire.com/download/7r4qy256aah675a/Character_Creation_1.1.rar

Also pls I'm a lazy asshole who doesn't even actually do anything at all.

**Comment by Incoming5643** [comment]
Alright sprites updated, wagging reintroduced.

Merge whenever!

**Comment by Fox-McCloud** [comment]
Thanks a million, @WJohn!

:+1:  to this PR now.

File: code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -355,7 +355,9 @@
 		return
 	if("tail" in dna.species.mutant_bodyparts)
 		dna.species.mutant_bodyparts -= "tail"
+		dna.species.mutant_bodyparts -= "spines"
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/emote.dm]: I'd move the two spines parts under
``` if("spines" in dna.species.mutant_bodyparts)```

since at the moment you're adding spines whether they have them or not by the looks of things.

Comment (Incoming5643) [inline code/modules/mob/living/carbon/human/emote.dm]: there's a safety to remove the bodypart from being "expressed" if it's set to null or the designated "none" datum. The actual mutant_bodyparts doesn't change even if they choose not to have spines.

Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/emote.dm]: Right.

**Comment by Cheridan** [comment]
Welp, everything looks in order.

**Comment by RemieRichards** [comment]
You heard it here first folks, Overreacting and hissy fits get results!

**Comment by SconesC** [comment]
Oh boy.

**Comment by GunHog** [comment]
> You heard it here first folks, Overreacting and hissy fits get results!

Except with Dismemberment!

**Comment by RemieRichards** [comment]
@GunHog but I'm consulting Ricotez on that, and some of my code is even been recycled for it, rofl.

**Comment by whitefox08** [comment]
Fucking flew to close to the sun.

**Comment by RogerWilcoBeep** [comment]
and of course one of the supporters for this was mister fox mc-fucking-cloud

**Comment by Incoming5643** [comment]
Was it not your sin that trapped the unicorn? Even now the evil seed of what you've done germinates within you.

**Comment by Shadowlight213** [comment]
Time to meme in the necroed pr!

**Comment by Dorsisdwarf** [comment]
troomp
